test/state/requests.cpp

```diff
@@ -0,0 +1,33 @@
+// evmone: Fast Ethereum Virtual Machine implementation
+// Copyright 2024 The evmone Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "requests.hpp"
+#include <evmone_precompiles/sha256.hpp>
+
+namespace evmone::state
+{
+hash256 calculate_requests_hash(std::span<const Requests> block_requests_list)
+{
+    bytes requests_hash_list;
+    requests_hash_list.reserve(sizeof(bytes32) * block_requests_list.size());
+
+    for (const auto& requests : block_requests_list)
+    {
+        // TODO recent change in the spec, uncomment for devnet-5
+        //        if (requests.data.empty())
+        //            continue;
+
+        const bytes requests_bytes = requests.type + requests.data;
```

Comment:
This can be optimized to get rid of copying, if we represent `Requests` as just `struct Requests { bytes data; }` and put type as the first byte of `data` right away when collecting requests.

Let me know if it's worth it. I think not really, as it only affects blockchain test runner performance.

Reply:
I think this is worth it when all EIPs are implemented. So let's make an issue to track it.

Reply:
> I think this is worth it when all EIPs are implemented. So let's make an issue to track it.

https://github.com/ethereum/evmone/issues/1086

Reply:
> BTW, is there rationale for this double-hashing?

Good question, I couldn't find any in the EIP, other than previously it was a root of MPT and now is hash of flat list of hashes.
Asked on magicians https://ethereum-magicians.org/t/eip-7685-general-purpose-execution-layer-requests/19668/20